src/core/library.h

```diff
@@ -343,8 +343,10 @@ QuicCidNewRandomSource(
         *Data = PartitionID;
         Data++;
 
-        QuicCopyMemory(Data, Prefix, PrefixLength);
-        Data += PrefixLength;
+        if (PrefixLength) {
+            QuicCopyMemory(Data, Prefix, PrefixLength);
+            Data += PrefixLength;
+        }
```

Comment:
So, does clang not like passing `NULL, 0` to `memcpy`? `memcpy` is practically a `for` loop and if length is zero it would never read from the `NULL` being passed in. This effectively just adds another (irrelevant) NULL check IMO.

Reply:
Passing null to memcpy is UB, even if 0 length is passed. Standard 7.1.4